dev-docs/bidders/huaweiads.md

```diff
@@ -18,14 +18,34 @@ sidebarType: 1
 
 ### Note
 
-The Example Bidding adapter requires setup before beginning. Please contact us at <hwads@huawei.com>.
+The Example Bidding adapter requires setup and approval before implementation. Please contact us at <developereu@huawei.com>, also you can [submit a ticket online](https://developer.huawei.com/consumer/en/support/feedback/#/).
 
 1. The following parameters need to be registered on the HuaweiAds platform, and at the same time, the permission to access the server interface needs to be opened on the HuaweiAds platform.
 2. You can find ( publisherid, signkey, keyid ) on the platform after registration.
 3. You need to create your advertising creative on the platform and get the corresponding ( slotid, adtype ).
-4. We need OAID, GAID, IMEI (One must not be empty, you can collect OAID first, or use GAID) in our request, so when using prebidmobile, please add something like "TargetingParams.addUserData("oaid", "oaid-test") TargetingParams.addUserData("gaid", "gaid-test") TargetingParams.addUserData("imei", "imei-test")".
+4. We need OAID or GAID in our request (OAID: Open Advertising ID, GAID: Google Advertising ID, An OAID or GAID is a non-permanent device identifier, based on which personalized ads can be served to users on the premise of protecting user data privacy. In most cases, OAID is used for HMS phones and GAID is used for GMS phones.), GAID is collected by default, for OAID specifically for HMS phones, please call the static AdvertisingIdClient.getAdvertisingIdInfo(mContext) method to [obtain the OAID](https://developer.huawei.com/consumer/en/doc/development/HMSCore-Guides/identifier-service-obtaining-oaid-sdk-0000001050064988), then add the OAID to the request like "TargetingParams.addUserData(“oaid”, AdvertisingIdClient.Info info = AdvertisingIdClient.getAdvertisingIdInfo(getApplicationContext()).getId())".
 5. You can also send the clientTime to the HuaweiAds Adx server, like "TargetingParams.addUserData("clientTime", "2018-11-02 16:34:07.981+0800")". if not, it will use the prebid server time and zone.
 
+### Obtain the OAID
+
+```java
+// Add the OAID to user data in prebid initialization
+if ( isHmsAvailable() ) {
+    TargetingParams.addUserData(“oaid”, AdvertisingIdClient.Info info = AdvertisingIdClient.getAdvertisingIdInfo(getApplicationContext()).getId());
```

Comment:
```suggestion
    TargetingParams.addUserData("oaid", AdvertisingIdClient.Info info = AdvertisingIdClient.getAdvertisingIdInfo(getApplicationContext()).getId());
```

not sure if all IDEs sanitizes the `“` properly when  this code is copy and pasted